### ISSUE TITLE
Symfony 4.2 upgrade and migration to Guard

### ISSUE THREAD

**Issue by umpirsky** [symfony/symfony#35102]
In https://github.com/symfony/symfony/blob/master/UPGRADE-4.2.md it says:

`SimpleAuthenticatorInterface`, `SimpleFormAuthenticatorInterface`, `SimplePreAuthenticatorInterface`,
   `SimpleAuthenticationProvider`, `SimpleAuthenticationHandler`, `SimpleFormAuthenticationListener` and
   `SimplePreAuthenticationListener` have been deprecated. Use Guard instead.

There is no information about the migration process:

- How to change security configuration?
- Will sessions be preserved or it will log users out like https://github.com/symfony/symfony/issues/33473?
- How to disable this deprecated interfaces and force using of Guard?

There seem to be lack of upgrade information here. All I can find is https://symfony.com/doc/4.2/security/form_login_setup.html and https://symfony.com/doc/current/security/guard_authentication.htm, but I feel that https://github.com/symfony/symfony/blob/master/UPGRADE-4.2.md can be imroved with some before and after code snippets.

**Comment by umpirsky** [comment]
I added:

```
# security.yaml
security:
    # ...

    firewalls:
        main:
            # ...
            guard:
                authenticators:
                    - App\Security\LoginFormAuthenticator
```

But I still get deprecation messages.

Am I missing something?

**Comment by javiereguiluz** [comment]
@wouterj do you think we should add something about this in the docs ... or is it enough to add a better example in the UPGRADE guide? Thanks.

**Comment by wouterj** [comment]
I don't think there is much to do here. It's not a simple "set this to true". If you are using one of the `Simple*` classes/interfaces in Symfony 4, you would have to throw away that code and write the same logic in the Guard system. This cannot be written in any sort of simple "example".

We've already documentation on how to write custom Guard authenticators, so that should be enough. If anything, we should maybe reword "Use Guard instead." in the UPGRADE guide.

**Comment by umpirsky** [comment]
What if one does not use `Simple*` classes/interface? How to "Use Guard instead"?

**Comment by wouterj** [comment]
If someone doesn't use the `Simple*` classes/interfaces (including any third party bundles that are used for security), there is nothing to do.

**Comment by xabbuh** [comment]
I am closing here as I think there is nothing to change. If someone still feels that the upgrade file could be improved, please just open a pull request with the suggested changes and we can review the concrete wording.

**Comment by TomasVotruba** [comment]
For those coming here in 2025+ like me: **Do not upgrade to Guard, it was deprecated in Symfony 5.3**  -https://symfony.com/blog/new-in-symfony-5-3-guard-component-deprecation

It would only add you extra work you'll drop in a couple weeks.

Instead, upgrade at least to Symfony 5.1, that includes new (not sure if latest in Symfony 6, 7...) authentication system:
https://symfony.com/blog/new-in-symfony-5-1-updated-security-system
